feat: coding mechanic (wip)

RedVortexDev opened the pull request.

RedVortexDev committed:
feat: implement code execution

Co-Authored-By: BlazeMCworld <62186782+BlazeMCworld@users.noreply.github.com>
Co-Authored-By: UserUNP <39243708+UserUNP@users.noreply.github.com>

RedVortexDev committed:
fix: spaces in vararg params + cancelable lore

* Trigger leave event

Co-Authored-By: BlazeMCworld <62186782+BlazeMCworld@users.noreply.github.com>

RedVortexDev committed:
feat: if statements + code block menu cleanup

Co-Authored-By: BlazeMCworld <62186782+BlazeMCworld@users.noreply.github.com>

BlazeMCworld committed:
feat: implement variable system

RedVortexDev committed:
refactor: reorder code actions/events

RedVortexDev committed:
refactor: register actions with codeblock

RedVortexDev committed:
Merge branch 'feat/code-execution' into refactor/register-codeblock-separately

RedVortexDev committed:
refactor: register codeblock in setvar + move

RedVortexDev committed:
refactor: register codeblock separately (#45)

RedVortexDev committed:
refactor: rename conditional to condition

RedVortexDev committed:
feat: make sendmessage messages optional

RedVortexDev committed:
fix: no error when placing codeblock w/out starter

RedVortexDev committed:
feat: disable shiftclicking vals into items

RedVortexDev committed:
refactor: '_' for unused vars + upgrade to jdk 21

BlazeMCworld committed:
fix: various item changes
- coercing from items now gets the item name
- coercing to items, now create an item based on the minecraft id
- prevent putting renamed items into regular value slots
-

RedVortexDev committed:
refactor: remove unused class

RedVortexDev committed:
feat: tags & var tags

Co-Authored-By: BlazeMCworld <62186782+BlazeMCworld@users.noreply.github.com>

Chickensoup20 reviewed (commented):
i am poo

UserUNP reviewed (approved):
stuff to be done (as of now):
- [ ] static types for variables (default: Any)
- [ ] variable scopes
  - [x] Line
  - [x] Game
  - [ ] Save
- [x] variables for block tags
- [x] implement vanilla code items
- [x] event targets
- [ ] selections
- [ ] game values
- [ ] %target%
- [ ] %var
- [ ] functions
- [ ] processes

RedVortexDev committed:
Merge branch 'master' into feat/code-execution

RedVortexDev committed:
feat: rejoin event

RedVortexDev committed:
refactor: move action item classes

RedVortexDev committed:
fix: arguments in actionitem lore

+ support tags

RedVortexDev committed:
feat: setvar categories

RedVortexDev committed:
feat: rename cancelable to cancellable

RedVortexDev committed:
feat: unmark rejoin as cancellable

RedVortexDev committed:
fix: bottom row of setvar menu

RedVortexDev committed:
feat: open action menu by rightclicking codeblock

RedVortexDev committed:
feat: add optional hint for action items

RedVortexDev committed:
feat: null value

RedVortexDev committed:
fix: typo in null value description

RedVortexDev committed:
feat: left-click barrel to quick insert items

RedVortexDev committed:
fix: join event called before player reset

UserUNP committed:
fix: patch left arm swing triggering rclick event

UserUNP committed:
fix: patch left arm swing triggering rclick event

UserUNP committed:
feat: drop item player event + some patches

UserUNP committed:
fix: get rid of most IDE warnings

UserUNP committed:
fix: i forgo

BlazeMCworld committed:
feat: implement repeat statements

adds a repeat multiple times code block, and a general system for further repeat blocks
also includes a system for stopping code running for longer than a second at once

RedVortexDev committed:
refactor: use plugin logger

RedVortexDev committed:
refactor: remove some warns

RedVortexDev committed:
feat: plot unloading (wip)

RedVortexDev committed:
Merge branch 'master' into feat/code-execution

RedVortexDev committed:
Merge branch 'master' into feat/code-execution

RedVortexDev committed:
fix: cancelable instead of cancellable

RedVortexDev committed:
fix: plot location for movement check

RedVortexDev committed:
feat: finish plot unloading

RedVortexDev committed:
refactor: reformat project

RedVortexDev committed:
Merge branch 'feat/code-execution' into feat/plot-unloading

RedVortexDev committed:
feat: plot unloading (#56)

UserUNP committed:
Merge branch 'master' into feat/code-execution

UserUNP commented:
fixed the merge conflict

RedVortexDev committed:
fix: set variable actionmenu not opening

RedVortexDev committed:
feat: global variables

RedVortexDev committed:
Merge branch 'feat/code-execution' of https://github.com/HypersquareDev/Hypersquare into feat/code-execution

RedVortexDev committed:
feat: other variable scope names

RedVortexDev committed:
chore: update .gitignore

RedVortexDev committed:
fix: leave doesn't trigger on player disconnect

RedVortexDev committed:
refactor: document hashmaps & listener registering

RedVortexDev committed:
refactor: remove left-over event cache

RedVortexDev committed:
feat: async per plot code exec & wait codeblock

Co-Authored-By: oven <39243708+UserUNP@users.noreply.github.com>

UserUNP committed:
chore: delete .settings directory

UserUNP committed:
chore: .gitignore

UserUNP committed:
wip(Targets): unfinished changes

RedVortexDev committed:
Merge branch 'master' into feat/code-execution

RedVortexDev committed:
style(var): var scope in name

RedVortexDev committed:
fix: lost cancellable fix in merge

RedVortexDev committed:
feat: implement all event listeners

RedVortexDev committed:
feat: start on implementing events

Chickensoup20 committed:
feat: added all missing events*

RedVortexDev committed:
feat: added all missing events (#58)

RedVortexDev committed:
feat: targets (wip)

UserUNP committed:
bruh

UserUNP committed:
fuck

UserUNP committed:
fuck2

UserUNP committed:
fuck3 (finale)

RedVortexDev committed:
fix: gitignore ignoring all folders named target

RedVortexDev committed:
refactor: rename targets to target

RedVortexDev committed:
fix: quick-inserted item saves stack size

RedVortexDev committed:
fix: targets tries to open when clicking codeblock

UserUNP committed:
fix(Target): make targets work as intended

Chickensoup20 committed:
fix: plot limit exceeded on west side

Chickensoup20 committed:
fix: death message and respawning at spawn

Chickensoup20 committed:
feat: Location Value (WIP)

Chickensoup20 committed:
feat: Location Value (WIP)

Chickensoup20 committed:
fix: Entities being teleported incorrectly

Chickensoup20 committed:
feat: Location Value

Co-Authored-By: RedVortex <58099979+RedVortexDev@users.noreply.github.com>

RedVortexDev committed:
refactor: use Colors class for some unported parts

UserUNP committed:
feat: HSException

Chickensoup20 committed:
Merge branch 'feat/code-execution' into fix/plot-limits

Chickensoup20 committed:
Fix/plot limits (#61)

UserUNP committed:
fix: wait action & jetbrains annotations & warnings

Chickensoup20 committed:
fix: being able to place water outside a plot

Chickensoup20 committed:
fix: blocks indirectly being placed outside plot

Chickensoup20 committed:
fix: errors using HSException

Chickensoup20 committed:
fix: moved plot related events to PlotEvents

RedVortexDev committed:
Fix/blocks escaping plot (#62)

RedVortexDev committed:
refactor(itembuilder): remove string methods

RedVortexDev committed:
feat: click events in values & var scope switch menu

RedVortexDev committed:
Merge branch 'feat/code-execution' of https://github.com/HypersquareDev/Hypersquare into feat/code-execution

RedVortexDev committed:
fix: setting value not keeping amount

RedVortexDev committed:
fix: setting values dont save stacksize & null value canceling messages (#63)

RedVortexDev committed:
fix: null value cancels chat messages when held

RedVortexDev committed:
refactor: a lot of changes (check description)

1. Fix claiming plot menu not opening
2. Make it so actions are not forced to have a category
3. New codeblock: Dev Action
4. New action: Print Stacktrace (Dev)
5. Refactor right-click logic
6. General refactoring
7. Special action type marker - currently advanced (unused) and dev.
8. HSLocation
9. Make it so wait doesn't use a category anymore
10. Clean up ValueCommands
11. Fix placing waits in repeats

RedVortexDev committed:
fix: notValid is not inverted

RedVortexDev committed:
refactor: remove debug

RedVortexDev committed:
fix: repeats iters once, event cancelling & renames

and remove thread scope.

RedVortexDev committed:
refactor: any codeblocks can have params & overall refactoring

Co-Authored-By: oven <39243708+UserUNP@users.noreply.github.com>

RedVortexDev committed:
Merge branch 'master' into feat/code-execution

RedVortexDev committed:
feat: thread starters save arguments and tags

Co-Authored-By: oven <39243708+UserUNP@users.noreply.github.com>

RedVortexDev committed:
Merge branch 'feat/code-execution' of https://github.com/HypersquareDev/Hypersquare into feat/code-execution

RedVortexDev committed:
fix: checking empty default action is incorrect

RedVortexDev committed:
refactor: remove some errors

RedVortexDev committed:
refactor: remove some errors (#64)

RedVortexDev reviewed (commented):
warnings*

RedVortexDev committed:
style: nicer exceptions

RedVortexDev committed:
fix: cant set action if clicked on codeblock

& remove delay

RedVortexDev committed:
Merge branch 'feat/code-execution' of https://github.com/HypersquareDev/Hypersquare into feat/code-execution

RedVortexDev committed:
fix: leave event code isn't executed

RedVortexDev committed:
fix: leaving server doesn't call leave event

Chickensoup20 committed:
feat: some more events

UserUNP committed:
feat: some more events (#65)

Chickensoup20 committed:
feat: teleport action

Chickensoup20 committed:
feat: display locations or errors

Chickensoup20 committed:
feat: made locations clickable to set location

Chickensoup20 committed:
fix: decimal point being negative

CNK12321 committed:
feat: <-- LOl FEET
+ Clear Inventory
+ Remove Item(s)
+ Gamemode Creative -> Set Gamemode
+ Added BarrelMenu#parameterRange so you don't have to type a millon billon slots!

CNK12321 committed:
feat:
+ Also made it say no parameters if there isn't any.

CNK12321 committed:
fix: Lore of ClearInv and Gamemode, and undid no param lore.

CNK12321 committed:
feat: More inventory actions, getNullable fixed.
+ New specific error type,
+ Player set hotbar items
+ Player set inventory items
+ Player set inventory slot
Changed the actions to use Action#getName instead of repeating it twice.

CNK12321 committed:
Request: Removed comment

CNK12321 committed:
Fix: Changed null to air and changed varargs in addAdditionalInfo to multiple calls of addAdditionalInfo

Sawbez committed:
Add basic player stat changes

Chickensoup20 committed:
feat: finished locations

Chickensoup20 committed:
Feat/finishing location (#69)

no obvious conflicts

Chickensoup20 committed:
feat: Player launch action

Chickensoup20 committed:
feat: Player boost elytra

Chickensoup20 committed:
feat: Player set gliding

Chickensoup20 committed:
feat: Player set flying

Chickensoup20 committed:
feat: more actions

Chickensoup20 committed:
feat: Face location action

Chickensoup20 committed:
feat: Player set rotation

Chickensoup20 committed:
feat: fixed location being wrong location

Chickensoup20 committed:
Merge branch 'feat/code-execution' into feat/code-execution

RedVortexDev committed:
Merge branch 'feat/code-execution' into feat/code-execution

RedVortexDev committed:
feat: basic player stat changes (#67)

RedVortexDev committed:
Merge branch 'feat/code-execution' into feat/code-execution

RedVortexDev committed:
feat: action additions (#66)

Chickensoup20 committed:
fix: fall distance and potion effects were not being reset

Chickensoup20 committed:
fix: if the server was lagging plots could not be created

Chickensoup20 committed:
fix: my plots menu size not changing with the amount of plots

Chickensoup20 committed:
fix: sent uuid of owner instead of name when joining a plot

Chickensoup20 committed:
Merge branch 'feat/code-execution' into feat/movement-actions

Chickensoup20 committed:
Feat/movement actions (#70)

Chickensoup20 committed:
fix: if you dont have any plots the menu breaks

Chickensoup20 committed:
Fix/various bug fixes (#72)

Soltrubeg committed:
impl: player respawn event

Soltrubeg committed:
feat: new events

Soltrubeg committed:
feat: new events (#73)



    Player Respawn Event
    Player Take Damage Event
    Player Damage Player Event
    Player Damage Entity Event
    Entity Damage Player Event

Soltrubeg committed:
feat: new player actions

Soltrubeg committed:
feat: new player actions and conditions

Soltrubeg committed:
feat: new player action

RedVortexDev committed:
Merge branch 'feat/code-execution' into feat/code-execution

Soltrubeg committed:
Update PlayerSendAttackAnimation.java

RedVortexDev committed:
new events & actions (#74)

Soltrubeg committed:
feat: player statistics

Soltrubeg committed:
feat: player world/appearance actions

Chickensoup20 committed:
Feat/code execution (#76)

RedVortexDev committed:
refactor: action & general code + param indicators

RedVortexDev committed:
feat: close inv action

RedVortexDev committed:
refactor: reorder Actions enum & end actions with Action

RedVortexDev committed:
refactor: action & general code + param indicators (#78)

RedVortexDev committed:
fix: None not showing up when no params & tags

RedVortexDev committed:
Merge branch 'feat/code-execution' of https://github.com/HypersquareDev/Hypersquare into feat/code-execution

RedVortexDev committed:
feat: replace 1 row empty invs with 3 row

RedVortexDev committed:
feat: center align tags

Soltrubeg committed:
feat: player launch projectile action

Soltrubeg committed:
feat: player launch projectile action

RedVortexDev committed:
Merge branch 'feat/code-execution' into feat/code-execution

RedVortexDev committed:
feat: player launch projectile action (#81)
